pkg/platform/kube/platform.go

```diff
@@ -1231,6 +1231,56 @@ func (p *Platform) enrichTriggerWithServiceType(functionConfig *functionconfig.C
 	return trigger
 }
 
+func (p *Platform) validateIngressHostAndPathAvailability(namespace, host, path string) error {
+	conflictErr := nuclio.NewErrConflict("Ingress host and path are already in use")
+	normalizedPath := common.NormalizeURLPath(path)
+
+	// get all ingresses on the namespace
+	ingresses, err := p.consumer.kubeClientSet.ExtensionsV1beta1().Ingresses(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "Failed to list ingresses")
+	}
+
+	// iterate over all ingresses to see if any of them matches host+path from args
+	for _, ingressInstance := range ingresses.Items {
+		for _, rule := range ingressInstance.Spec.Rules {
+			if host == rule.Host {
+
+				// if rule HTTP struct is nil - return conflict error only if the path is empty
+				if rule.HTTP == nil {
+					if normalizedPath == "/" {
+						return conflictErr
+					}
+
+					// rule host HTTP struct is nil - continue to the next rule
+					continue
+				}
+
+				// check if one of the paths matches our path
+				for _, pathInstance := range rule.HTTP.Paths {
+					if normalizedPath == common.NormalizeURLPath(pathInstance.Path) {
+						return conflictErr
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (p *Platform) validateAPIGatewayIngresses(apiGatewayConfig *platform.APIGatewayConfig) error {
+	if err := p.validateAPIGatewayFunctionsHaveNoIngresses(apiGatewayConfig); err != nil {
+		return errors.Wrap(err, "Failed while validating api gateway functions have no ingresses")
+	}
+
+	if err := p.validateIngressHostAndPathAvailability(apiGatewayConfig.Meta.Namespace, apiGatewayConfig.Spec.Host, apiGatewayConfig.Spec.Path); err != nil {
```

Comment:
simply pass the `apiGatewayConfig` structure, future prof for changes when requiring more fields

Reply:
But then we couldn't share this function is between api gateways and functions